### ISSUE TITLE
Add missing semantics to licenses page

### ISSUE THREAD

**Issue by kennethj** [flutter/flutter#84441]
## Issue
When using a screen reader, `ListTile`s that navigate to license details are not announced as interactive. It is not intuitive that these elements reveal more information.

## Proposal
Announce these elements as buttons or links using the `Semantics` widget.

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/about.dart#L665-L728
